### PR TITLE
chore: bump PowerTools Java version to 1.16.1

### PR DESCRIPTION
**Pull request by AlexeySoshin**
*Issue #, if available:*
https://github.com/aws-powertools/powertools-lambda-java/issues/1278

*Description of changes:*
Bumping PowerTools Java version to 1.16.0, latest available 
Bump Gradle to 7.6.2, latest Gradle 7 version available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

*Test*
- [x] Java 17 Gradle
- [x] Java 17 Maven
- [x] Java 11 Maven
- [x] Java 8 Maven
- [x] Java 8 AL Maven

### CODE REVIEW

File: java17/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml [hunk 1]
```diff
@@ -16,21 +16,21 @@
         <dependency>
             <groupId>software.amazon.lambda</groupId>
             <artifactId>powertools-tracing</artifactId>
-            <version>1.13.0</version>
+            <version>1.16.0</version>
```
Comment (jeromevdl) [inline java17/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml:19]:
```suggestion
            <version>1.16.1</version>
```

File: java17/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml [hunk 2]
```diff
@@ -16,21 +16,21 @@
...
         </dependency>
         {%- endif %}
         {%- if cookiecutter[ "Powertools for AWS Lambda (Java) Logging" ] == "enabled" %}
         <dependency>
             <groupId>software.amazon.lambda</groupId>
             <artifactId>powertools-logging</artifactId>
-            <version>1.13.0</version>
+            <version>1.16.0</version>
```
Comment (jeromevdl) [inline java17/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml:26]:
```suggestion
            <version>1.16.1</version>
```

File: java17/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml [hunk 3]
```diff
@@ -16,21 +16,21 @@
...
         </dependency>
         {%- endif %}
         {%- if cookiecutter[ "Powertools for AWS Lambda (Java) Metrics" ] == "enabled" %}
         <dependency>
             <groupId>software.amazon.lambda</groupId>
             <artifactId>powertools-metrics</artifactId>
-            <version>1.13.0</version>
+            <version>1.16.0</version>
```
Comment (jeromevdl) [inline java17/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml:33]:
```suggestion
            <version>1.16.1</version>
```

File: java8.al2/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml [hunk 4]
```diff
@@ -16,21 +16,21 @@
         <dependency>
             <groupId>software.amazon.lambda</groupId>
             <artifactId>powertools-tracing</artifactId>
-            <version>1.13.0</version>
+            <version>1.16.0</version>
```
Comment (jeromevdl) [inline java8.al2/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml:19]:
```suggestion
            <version>1.16.1</version>
```

File: java8.al2/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml [hunk 5]
```diff
@@ -16,21 +16,21 @@
...
         </dependency>
         {%- endif %}
         {%- if cookiecutter[ "Powertools for AWS Lambda (Java) Logging" ] == "enabled" %}
         <dependency>
             <groupId>software.amazon.lambda</groupId>
             <artifactId>powertools-logging</artifactId>
-            <version>1.13.0</version>
+            <version>1.16.0</version>
```
Comment (jeromevdl) [inline java8.al2/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml:26]:
```suggestion
            <version>1.16.1</version>
```

File: java8.al2/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml [hunk 6]
```diff
@@ -16,21 +16,21 @@
...
         </dependency>
         {%- endif %}
         {%- if cookiecutter[ "Powertools for AWS Lambda (Java) Metrics" ] == "enabled" %}
         <dependency>
             <groupId>software.amazon.lambda</groupId>
             <artifactId>powertools-metrics</artifactId>
-            <version>1.13.0</version>
+            <version>1.16.0</version>
```
Comment (jeromevdl) [inline java8.al2/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml:33]:
```suggestion
            <version>1.16.1</version>
```

File: java8/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml [hunk 7]
```diff
@@ -16,21 +16,21 @@
         <dependency>
             <groupId>software.amazon.lambda</groupId>
             <artifactId>powertools-tracing</artifactId>
-            <version>1.13.0</version>
+            <version>1.16.0</version>
```
Comment (jeromevdl) [inline java8/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml:19]:
```suggestion
            <version>1.16.1</version>
```

File: java8/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml [hunk 8]
```diff
@@ -16,21 +16,21 @@
...
         </dependency>
         {%- endif %}
         {%- if cookiecutter[ "Powertools for AWS Lambda (Java) Logging" ] == "enabled" %}
         <dependency>
             <groupId>software.amazon.lambda</groupId>
             <artifactId>powertools-logging</artifactId>
-            <version>1.13.0</version>
+            <version>1.16.0</version>
```
Comment (jeromevdl) [inline java8/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml:26]:
```suggestion
            <version>1.16.1</version>
```

File: java8/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml [hunk 9]
```diff
@@ -16,21 +16,21 @@
...
         </dependency>
         {%- endif %}
         {%- if cookiecutter[ "Powertools for AWS Lambda (Java) Metrics" ] == "enabled" %}
         <dependency>
             <groupId>software.amazon.lambda</groupId>
             <artifactId>powertools-metrics</artifactId>
-            <version>1.13.0</version>
+            <version>1.16.0</version>
```
Comment (jeromevdl) [inline java8/hello-pt-maven/{{cookiecutter.project_name}}/HelloWorldFunction/pom.xml:33]:
```suggestion
            <version>1.16.1</version>
```

**Review by jeromevdl — COMMENTED** [review]
we released a 1.16.1 with bug fixes yesterday. otherwise, it looks good (thanks for the Java updates: multiline in 17, var in 11...)

**Comment by jeromevdl** [comment]
@AlexeySoshin did you try to generate and deploy these ? Would be great to make sure everything works properly. Using [`sam init -l`](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-init.html#sam-cli-command-reference-sam-init-options) you can use a local file/zip to test.

**Comment by jeromevdl** [comment]
Don't know why python is failing... (we only modified java)

**Comment by AlexeySoshin** [comment]
@jeromevdl I did generate, run locally and deploy all 5 of these.
Also made sure that metrics and logs are actually collected.

**Review by mildaniel — APPROVED** [review]

**Review by mndeveci — APPROVED** [review]

**Pull request merged**